### PR TITLE
[new opmath] Simplify `LinearCombination(coeffs, ops, simplify=True)`

### PR DESCRIPTION
**Pull request by Qottmann**
From a [suggestion](https://github.com/PennyLaneAI/pennylane/pull/5216#discussion_r1530800753) in the original PR.

[sc-59440]

### CODE REVIEW

**Comment by astralcai** [comment]
This is indeed very strange 🤔

File: pennylane/ops/op_math/linear_combination.py [hunk 1]
```diff
@@ -275,44 +280,45 @@ def wires(self):
...
+        if len(ops) == 1:
+            return coeffs, [ops[0].simplify()], pr
 
-        op_as_sum = qml.sum(*self.operands)
+        op_as_sum = qml.dot(coeffs, ops)
         op_as_sum = op_as_sum.simplify(cutoff)
-        coeffs, ops = op_as_sum.terms()
-        return coeffs, ops, None
+        new_coeffs, new_ops = op_as_sum.terms()
+        return coeffs, ops, pr
```
Comment (astralcai) [inline pennylane/ops/op_math/linear_combination.py:316]: I'm assuming you meant this? Unless I'm missing something.
```suggestion
        return new_coeffs, new_ops, pr
```

**Review by astralcai — COMMENTED** [review]
I've figured out the issue. When a `CompositeOp` is constructed, `CompositeOp.queue()` removes the ops within the composite op and queue the composite op instead. In the old approach, this step was performed before simplification, but with the new approach, `_simplify_coeffs_ops` returns a new list of ops, which we assign to `operands`, so when we call `super().__init__()`, the list of `operands` is not the same as the previous ops that the composite op consisted of, so they are never removed from the queue, so we end up with an additional `X(0)` and a `Y(0)` in the active context.

**Comment by Qottmann** [comment]
@astralcai how sure are you that this is causing the reverse minus sign? I just pushed a version that simply skips the CompositeOp.__init__ but the reversed signs problem persists

**Comment by Qottmann** [comment]
Thanks @astralcai you were right. I now simply manually remove the initial `ops` in case `simplify=True`. This also restores differentiability using `torch`, so I think this is overall a win.

**Comment by Qottmann** [comment]
I get failures for tests using legacy opmath in qaoa now. Can that be related? I would think not, though tests seem to pass on the lower branch `ham-tests` so it must be something in this PR 🤔

**Review by astralcai — APPROVED** [review]
Looks good! 🎸

File: pennylane/ops/op_math/linear_combination.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 """
 LinearCombination class
 """
-# pylint: disable=too-many-arguments, protected-access
+# pylint: disable=too-many-arguments, protected-access, super-init-not-called, too-many-instance-attributes
```
Comment (mudit2812) [inline pennylane/ops/op_math/linear_combination.py:17]: Why did the `super-init-not-called` warning have to be disabled? `super().__init__` is being called on line 149

**Review by mudit2812 — APPROVED** [review]
Looks good! Just one comment about linting, but not blocking merging.

File: pennylane/ops/op_math/linear_combination.py (hunk 2 above)
Comment (Qottmann) [reply pennylane/ops/op_math/linear_combination.py:17]: Remnant from a previous attempt, good catch!

**Review by albi3ro — APPROVED** [review]

**Comment by trbromley** [comment]
> Simplify LinearCombination(coeffs, ops, simplify=True)

I'm confused by the title :laughing: Is this PR just simplifying how things work behind the scenes when `simplify is True`?

**Comment by Qottmann** [comment]
> I'm confused by the title 😆 Is this PR just simplifying how things work behind the scenes when simplify is True?

Fair! 😆 
Yes it is simplifying the logic of the `__init__` when the kward `simplify=True`. It also restores differentiability of the coeffs when using the torch interface (this was broken before).

**Pull request merged**